dialogy/types/entity/time/__init__.py

```diff
@@ -189,7 +192,21 @@ def set_entity_type(self) -> None:
             self.entity_type = const.TIME
 
     @classmethod
-    def from_duckling(cls, d: Dict[str, Any], alternative_index: int) -> TimeEntity:
+    def pick_constrained_duckling_entities(cls, d: Dict[str, Any], constraints: Dict[str, Any]) -> Dict[str, Any]:        
+        grain = d[const.VALUE][const.VALUES][0][const.GRAIN]
```

Comment:
The PR looks good but this section is very hard to read. 
An untrained pair of eyes would venture here to fix a bug and will potentially break something else on their way out.